server/chat-plugins/poll.ts

```diff
@@ -350,27 +350,33 @@ export const commands: ChatCommands = {
 
 		clearqueue: 'deletequeue',
 		deletequeue(target, room, user, connection, cmd) {
+			const [slotString, roomid, update] = target.split(',').map(i => i.trim());
+			const targetRoom = roomid ? Rooms.search(roomid) : room;
+			if (!targetRoom) return this.errorReply(`Invalid room.`);
+			if (!targetRoom.persist) {
+				return this.errorReply("This command is unavailable in temporary rooms.");
+			}
+			this.room = targetRoom;
```

Comment:
Additionally, set `room = targetRoom`, then change all the uses of `targetRoom` back to `room`. Cleaner that way.